docs/en/engines/database-engines/overlay.md

```diff
@@ -0,0 +1,167 @@
+
+-- Create the overlay facade
+CREATE DATABASE dboverlay ENGINE = Overlay('db_overlay_a', 'db_overlay_b');
+
+-- Discover and read through overlay
+SHOW TABLES FROM dboverlay;                -- t_a, t_b
+SELECT * FROM dboverlay.t_a ORDER BY id;   -- rows from db_overlay_a.t_a
+SELECT * FROM dboverlay.t_b ORDER BY id;   -- rows from db_overlay_b.t_b
+
+-- Add a new table in an underlying DB (overlay is read-only for DDL)
+CREATE TABLE db_overlay_a.t_new (k UInt32, v String) ENGINE = MergeTree ORDER BY k;
+INSERT INTO db_overlay_a.t_new VALUES (100,'x'), (200,'y');
+
+-- Read the new table via overlay
+SHOW TABLES FROM dboverlay;                -- now includes t_new
+SELECT * FROM dboverlay.t_new ORDER BY k;  -- rows from db_overlay_a.t_new
+
+-- Rename/drop in the underlying DB; overlay reflects changes
+RENAME TABLE db_overlay_a.t_new TO db_overlay_a.t_new_renamed;
+SELECT * FROM dboverlay.t_new_renamed ORDER BY k;
+
+DROP TABLE db_overlay_a.t_new_renamed;
+SHOW TABLES FROM dboverlay;                -- t_new_renamed disappears
+
+-- Remove the overlay (does not touch member DBs)
+DROP DATABASE dboverlay SYNC;
+```
+
+---
+
+## Troubleshooting {#troubleshooting}
+
+| Symptom                                                                               | Cause                                                                     | Fix                                                                                                                                |
+| :------------------------------------------------------------------------------------ | :------------------------------------------------------------------------ | :--------------------------------------------------------------------------------------------------------------------------------- |
+| `Logical error: 'Mapping for table with UUID=… already exists'` when creating overlay | Overlay DB had a non-Nil UUID colliding with existing DB/table            | In facade mode, **return Nil** from `getUUID()` (you already do).                                                                  |
+| `TABLE_ALREADY_EXISTS` under `store/000/00000000-…`                                   | Table was created **through** overlay; its storage bound to Nil-UUID path | Block `CREATE TABLE` via overlay; create in a member DB instead. Remove stale Nil dir if created.                                  |
+| `DATABASE_NOT_EMPTY` while dropping overlay                                           | Dropper saw tables or `empty()` returned false                            | In facade mode: return **empty iterator** for `skip_not_loaded==true` and **true** from `empty()`. Make `dropTable()` a **no-op**. |
+| Segfault in `canContainMergeTreeTables()` (or similar)                                | `databases` contained a **null** member                                   | Enforce non-null on registration; add `if (db)` guards in all loops (done).                                                        |
```

Comment:
Looks like all fixes already applied in the code, and I think this should be either some comments in the code or a comment in PR/commit